bot/exts/discord/discord.py

```diff
@@ -36,6 +43,49 @@ async def avatar(self, inter: ApplicationCommandInteraction, user: User) -> None
 
         await inter.response.send_message(embed=embed)
 
+    @discord.sub_command()
+    async def server(
+        self,
+        inter: ApplicationCommandInteraction,
+        guild_id: Optional[Guild] = None,
+        invite: Optional[Invite] = None,
+    ) -> None:
+        """
+        Get the icon of a Discord server.
+
+        Parameters
+        ----------
+        guild_id: The ID of the Discord server
+        invite: The invite to the Discord server
+        """
+        match (guild_id, invite):
+            case (None, None):
+                guild = inter.guild
+                icon = guild.icon
+            case (None, invite):
+                if (guild := invite.guild) is None:
+                    raise BadArgument("Cannot fetch the icon of a group DM.")
+                icon = guild.icon
+            case (guild, _):
+                # Only `guild_id` or, `guild_id` and `invite` specified.
+                icon = guild.icon
+
+        if icon is None:
+            embed = create_embed(
+                embed_type="warning",
+                title="Unable to fetch server icon.",
+                description="The server does not have an icon.",
+            )
+        else:
+            embed = create_embed(
+                title="Discord server icon",
+                description=f"Showing server icon of `{guild.name}`.",
+                fields={"Link": f"[Download icon]({icon})"},
+            )
+            embed.set_thumbnail(icon)
+
+        await inter.response.send_message(embed=embed, ephemeral=icon is None)
```

Comment:
I think that `ephemeral=icon is None` is a bit unclear here. What about making a variable like `is_ephemeral`, `error` or similar here?

Reply:
Is it really that unclear? Sure, I can set `is_ephemeral` to `True` in the if-conditional block.

Reply:
I think what I find unclear here is why `icon is None` actually represents. There being no icon doesn't tell me that an error has occurred, which is why the message is really being shown here.

Reply:
I've decided to raise an error if `icon` is `None` and let the (currently non-existent) error-handler handle it.